app/controllers/category_experts_controller.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class CategoryExpertsController < ApplicationController
+
+  def endorse
+    raise Discourse::NotFound unless current_user
+    user = fetch_user_from_params
+
+    category_ids = params[:categoryIds]&.reject(&:blank?)
+
+    raise Discourse::InvalidParameters if !category_ids || category_ids.blank?
+
+    categories = Category.where(id: category_ids)
+    categories.each do |category|
+      raise Discourse::InvalidParameters unless category.accepting_category_expert_endorsements?
+
+      CategoryExpertEndorsement.find_or_create_by(user: current_user, endorsed_user: user, category: category)
+    end
+
+    render json: {
```

Comment:
I noticed in the video that the user doesn't receive any kind of "success" message in the UI when this works. Maybe the window could briefly show a thumbs up or something before disappearing?

Reply:
https://share.getcloudapp.com/z8uPBz66

Reply:
Definitely an improvement there! We might want to adjust with feedback but much better than before.